### PR TITLE
Document ability to include functions in context

### PR DESCRIPTION
**Pull request by djmitche**
```js
let slugid = require('slugid');

let context = {
  slugid: () => slugid.nice()
};
let template = { .. {$eval: 'slugid()'}, .. }
console.log(jsone(context, template));

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -241,3 +241,31 @@ of the context.
...
+
+var template = {
+  title: "Trip to Hawaii",
+  thumbnail: {$eval: 'imageData("hawaii")`},
+};
+
+return jsone(template, context);
+```
+
+NOTE: Context functions are called synchrnously.  Any complex asynchronous
```
Comment (TheNavigat) [inline README.md]: nit: Extra space before `Any`

File: README.md [hunk 2]
```diff
@@ -241,3 +241,31 @@ of the context.
...
+operations should be handled before rendering the template.
+
+NOTE: If the template is untrusted, it can pass arbitrary data to functions in
```
Comment (TheNavigat) [inline README.md]: 79 characters. (I'm not sure if the limit was 78 or 80, so if it's > 78, ignore this)

**Comment by djmitche** [comment]
I don't think we really have a line-length limit in .md files (since they get rendered with line-wrap) but I'll fix it up.

**Comment by TheNavigat** [comment]
Yep I don't think so too but I've always abode (it's abode, isn't it?) by the "78 characters max" rule except when it's HTML. Short lines are nice, even if it's md :)

**Comment by djmitche** [comment]
Haha, that's a tough English question.  I think it's "abided by" :)

**Comment by TheNavigat** [comment]
abided shows a read underline in Chrome. Abode is correct, https://www.merriam-webster.com/dictionary/abide. Apparently abided is also correct anyway :)

**Comment by djmitche** [comment]
conflicts resolved -- @TheNavigat can you have another look?

**Pull request merged**